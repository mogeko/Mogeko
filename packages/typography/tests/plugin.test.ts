import { describe, it, expect } from "vitest";
import plugin from "@/index";
import tailwindcss from "tailwindcss";
import postcss from "postcss";
import path from "path";

describe("typography plugin", () => {
  it("default styles", async () => {
    const { currentTestName } = expect.getState();
    const result = await postcss(
      tailwindcss({
        content: [{ raw: String.raw`<div class="prose lg:prose-lg"></div>` }],
        plugins: [plugin],
        corePlugins: { preflight: false },
      }),
    ).process(["@tailwind base;", "@tailwind components;"].join("\n"), {
      from: `${path.resolve(__filename)}?test=${currentTestName}`,
    });

    expect(result.css).toMatchSnapshot();
  });

  it("with custom className", async () => {
    const className = "custom";

    const { currentTestName } = expect.getState();
    const result = await postcss(
      tailwindcss({
        content: [{ raw: String.raw`<div class="${className}"></div>` }],
        plugins: [plugin({ className })],
        corePlugins: { preflight: false },
      }),
    ).process(["@tailwind base;", "@tailwind components;"].join("\n"), {
      from: `${path.resolve(__filename)}?test=${currentTestName}`,
    });

    expect(result.css).toMatch(`.${className} h1`);
    expect(result.css).not.toMatch(`.${className}-lg h1`);
  });

  it("with custom styles", async () => {
    const { currentTestName } = expect.getState();
    const result = await postcss(
      tailwindcss({
        content: [{ raw: String.raw`<div class="prose lg:prose-lg"></div>` }],
        plugins: [plugin],
        corePlugins: { preflight: false },
        theme: {
          typography: {
            DEFAULT: { h1: { fontSize: "1.5rem" } },
            lg: { h1: { fontSize: "2rem" } },
          },
        },
      }),
    ).process(["@tailwind base;", "@tailwind components;"].join("\n"), {
      from: `${path.resolve(__filename)}?test=${currentTestName}`,
    });

    expect(result.css).toMatch("font-size: 1.5rem");
    expect(result.css).toMatch("font-size: 2rem");
    expect(result.css).not.toMatch(".prose h1:not(:first-child)");
  });
});
