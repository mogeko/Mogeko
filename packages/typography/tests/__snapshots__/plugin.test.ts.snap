// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`typography plugin > default styles 1`] = `
"*, ::before, ::after {
    --tw-border-spacing-x: 0;
    --tw-border-spacing-y: 0;
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-gradient-from-position:  ;
    --tw-gradient-via-position:  ;
    --tw-gradient-to-position:  ;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  ;
}
::backdrop {
    --tw-border-spacing-x: 0;
    --tw-border-spacing-y: 0;
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-gradient-from-position:  ;
    --tw-gradient-via-position:  ;
    --tw-gradient-to-position:  ;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  ;
}
.prose h1 {
    font-size: 2.25rem;
    font-weight: 900;
    letter-spacing: -0.025em;
    line-height: 2.5rem;
    scroll-margin: 5rem;
}
.prose h1:not(:first-child) {
    margin: 3.5rem 0 1.75rem;
}
@media (min-width: 1024px) {
    .lg\\:prose-lg h1 {
        font-size: 3rem;
        line-height: 1;
    }
}"
`;
